projects/centre_console/src/race_switch.c

```diff
@@ -0,0 +1,125 @@
+#include "race_switch.h"
+
+#include "centre_console_events.h"
+#include "event_queue.h"
+#include "fsm.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "status.h"
+#include "voltage_regulator.h"
+
+#define VOLTAGE_REGULATOR_ENABLE_ADDRESS \
+  { .port = GPIO_PORT_B, .pin = 6 }
+#define VOLTAGE_REGULATOR_MONITOR_ADDRESS \
+  { .port = GPIO_PORT_B, .pin = 7 }
+#define VOLTAGE_REGULATOR_DELAY_MS 25
+
+static GpioAddress s_race_switch_address = { .port = GPIO_PORT_A, .pin = 4 };
+
+// Must be redefined to pass as a pointer
+static GpioAddress s_voltage_regulator_enable_address = VOLTAGE_REGULATOR_ENABLE_ADDRESS;
+static GpioAddress s_voltage_regulator_monitor_address = VOLTAGE_REGULATOR_MONITOR_ADDRESS;
+
+// Gpio state is low since car begins in normal mode
+static GpioSettings s_gpio_settings = { .state = GPIO_STATE_LOW,
+                                        .direction = GPIO_DIR_IN,
+                                        .resistor = GPIO_RES_NONE,
+                                        .alt_function = GPIO_ALTFN_NONE };
+
+static InterruptSettings s_interrupt_settings = { .type = INTERRUPT_TYPE_INTERRUPT,
+                                                  .priority = INTERRUPT_PRIORITY_NORMAL };
+
+static void prv_voltage_monitor_error_callback(VoltageRegulatorError error, void *context) {
+  LOG_WARN("Voltage Regulator Error %d\n", error);
```

Comment:
Maybe to save debugging time in the future you could print "off when should be on" or "on when should be off" as appropriate according to the value of `error` instead of the raw number